lib/dependable.rb

```diff
@@ -0,0 +1,19 @@
+module Dependable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :records_of_dependent_editions, class_name: 'EditionDependency', as: :dependable, dependent: :destroy
+    has_many :dependent_editions, through: :records_of_dependent_editions, source: :edition
+  end
+
+  def republish_dependent_editions
+    dependent_editions.each { |e| Whitehall::PublishingApi.republish(e) }
+  end
+
+  def destroy_records_of_dependent_editions
```

Comment:
I don't think that this method belongs in the `Dependable` module because it is not relevant to all dependable things. It's only relevant to `Editions`. I think it would be more appropriate to put this on the `Edition` class